app/static/js/createEvents.js

```diff
@@ -34,16 +34,47 @@ function format24to12HourTime(timeStr){
     }
     return formattedTime;
   }
+
+  function calculateRecurringEventFrequency(){
+      var eventDatesAndName = {name:$("#inputEventName").val(),
+                               isRecurring: true,
+                               startDate:$("#startDatePicker").val(),
+                               endDate:$("#endDatePicker").val()}
+      $.ajax({
+        type:"POST",
+        url: "/makeRecurringEvents",
+        data: eventDatesAndName, //get the startDate, endDate and name as a dictionary
+        success: function(jsonData){
+          var recurringEvents = JSON.parse(jsonData)
+          var recurringTable = $("#recurringEventsTable")
+          $("#recurringEventsTable tbody tr").remove();
+
+          for (var event of recurringEvents){
+            var eventdate = new Date(event.date).toLocaleDateString()
+            recurringTable.append("<tr><td>"+event.name+"</td><td><input name='week"+event.week+"' type='hidden' value='"+eventdate+"'>"+eventdate+"</td></tr>");
+            }
+        },
+        error: function(error){
+          console.log(error)
+        }
+      });
+
+  }
 /*
  * Run when the webpage is ready for javascript
  */
 $(document).ready(function() {
+  if ( $("#startDatePicker").val() && $("#endDatePicker").val()){
+    calculateRecurringEventFrequency();
+  }
```

Comment:
Remove this check, it does not have any actual use and it breaks the code.

Reply:
Keep check but change it to check if the start and end dates are different instead